Feature/tailwind

coreymcollins opened the pull request:
Closes #545
Closes #546 
Closes #547

### DESCRIPTION ###
- Adds support for Tailwind
- Updates wd_s to strip out opinionated styles and elements
- Removes all ACF JSON/ACF Blocks functionality

### OTHER ###
- [x] Is this issue accessible? (Section 508/WCAG 2.0AA)
- [x] Does this issue pass all the linting? (PHPCS, ESLint, SassLint)
- [x] Does this pass CBT?

### STEPS TO VERIFY ###
1. Check out the branch
2. Run `npm i`
3. Off to the races!

### DOCUMENTATION ###
Yes, this is currently updated on my local awaiting for this to be approved before deploying.

coreymcollins committed:
Strips things down to basics and adds Tailwind

coreymcollins committed:
Updates the pot file script

coreymcollins committed:
Adds the down caret for parent menu items

coreymcollins committed:
Removes the container class from the parent site-main

coreymcollins committed:
Updates breakpoints and mobile padding

coreymcollins committed:
Updates packages

coreymcollins committed:
Reduces parent menu icon size

coreymcollins committed:
Removes some link styles

coreymcollins committed:
Adjusts site-main spacing globally in the config file

coreymcollins committed:
Moves template styles to their own group

coreymcollins committed:
Hides the updated byline date

coreymcollins committed:
Adds image class spacing

coreymcollins committed:
Adds basic table styles

coreymcollins committed:
Updates some scripts

coreymcollins committed:
Replaces some config settings with variables

coreymcollins committed:
Updates tailwind

coreymcollins committed:
Removes an empty space

coreymcollins committed:
Adds future settings for 2.0 readiness

coreymcollins committed:
Adjusts purging/layers and layer targeting

coreymcollins committed:
Adds the sidebar template with columns layout

coreymcollins committed:
Fixes a renaming typo

coreymcollins committed:
Update version numbers

asharirfan committed:
Leverage @wordpress/scripts as primary tool chain (#552)

* 📦 NEW: webpack config file

* 📦 NEW: imagemin-webpack-plugin to optimize images via wp-scripts

* Remove svgo config file

* 👌 IMPROVE: Config comments

* Squashed commit of the following:

Merges tailwind branch down to fix some conflicts

Co-authored-by: Corey Collins <coreymcollins@gmail.com>

coreymcollins committed:
Updates the readme

coreymcollins committed:
Package lock update

coreymcollins committed:
Removes some extraneous styles from the config

coreymcollins committed:
Adjusts heading tag margins

coreymcollins committed:
Adds the copyright year shortcode

coreymcollins committed:
Adds footer nav menu

coreymcollins committed:
Adds some base styles for a mobile menu

coreymcollins committed:
Updates some spacing

coreymcollins committed:
Adds mobile menu toggling

coreymcollins committed:
Adds a little comment spacing

coreymcollins committed:
Adds pagination and some additional template tags

coreymcollins committed:
Updates the readme

coreymcollins committed:
Adds helpers and functions

coreymcollins committed:
Adds some of our hooks back in

coreymcollins committed:
Adds default modal styles

coreymcollins committed:
Rearranges some template tag ordering to be in line with the wiki

coreymcollins committed:
Removes our old utilities partials so that we can be Tailwind-first

coreymcollins committed:
Updates the results not found page markup

coreymcollins committed:
Removes an extra container class

coreymcollins committed:
Adjusts mobile menu styles

coreymcollins committed:
Adjusts word break styles

coreymcollins committed:
Removes bae styles for config settings

itsamoreh commented on header.php:
```diff
@@ -25,37 +25,34 @@
 
 				<?php if ( is_front_page() && is_home() ) : ?>
-					<h1 class="site-title"><a href="<?php echo esc_url( home_url( '/' ) ); ?>" rel="home"><?php bloginfo( 'name' ); ?></a></h1>
+					<h1 class="site-title text-4xl"><a href="<?php echo esc_url( home_url( '/' ) ); ?>" class="no-underline" rel="home"><?php bloginfo( 'name' ); ?></a></h1>
```
Maybe this is a non-issue but if we ever have to update the Tailwind config, to change font sizes for example, unless we extend the default config or use the same names, these font sizes will be irrelevant.

Same goes for padding and colors. Especially colors. A mockup may not necessarily have "bg-gray-200".

Might be a non-issue because when building a new site, you'll have to come in and edit the header anyway and you can remove these at that time.

coreymcollins replied:
Good call – we should make it as easy as possible to get things going without headaches or confusion. I think it'd be good to replace these types of text classes with the heading classes we usually use (like here, probably an `h3` class) and give some more thought to any spots we're using backgrounds or padding.

itsamoreh commented on tailwind.config.js:
```diff
@@ -0,0 +1,598 @@
+	},
+	variants: {},
+	plugins: [
+		plugin( function({ addBase, config }) {
```
It might be worth extracting these plugins into separate files to reduce the size of the main config file.

itsamoreh commented on tailwind.config.js:
```diff
@@ -0,0 +1,598 @@
+				},
+			}
+
+			const menus = {
```
Is the expectation to create a plugin to style blocks or other elements that are added? Like if I created a card, should I come in and create a tailwind plugin for cards?

I really don't like this CSS-in-JS syntax. Can we do this with `.css` or `.scss` files instead and use `@apply` to add tailwind utility classes? That way you have `header.php` for the markup and `header.scss` for the styles.

coreymcollins replied:
The expectation right now is that the most base of base styles exist in the config, and styles specific to the theme/project would be added via Sass partials (or Tailwind classes in the markup, or whatever method is preferred). 

I'm not against moving these things out of the config file at all, though, if we'd rather have them live as regular Sass partials instead.

coreymcollins committed:
Moves post and comment styles to partials instead of the config

coreymcollins committed:
Moves some base elements out of config

coreymcollins committed:
Moves form styles

coreymcollins committed:
Moves menu styles

coreymcollins committed:
Moves alignment and some additional globals

coreymcollins committed:
Moves off-canvas styles

coreymcollins committed:
Moves modals into a partial

coreymcollins committed:
Moves table styles

coreymcollins committed:
Replaces some CSS with @apply rules

coreymcollins committed:
Updates border styles

coreymcollins committed:
Updates base font sizes and spacing values

coreymcollins committed:
Updates some spacing

coreymcollins committed:
Updates box shadow to use REM instead of pixel values

coreymcollins committed:
Updates global heading sizes

coreymcollins committed:
Moves header styles/classes

coreymcollins committed:
Moves search styles

coreymcollins committed:
Moves footer styles

coreymcollins committed:
Cleans up single post styles

coreymcollins committed:
Adjusts sidebar template styles

coreymcollins committed:
Updates the caret class and purges JS files

coreymcollins committed:
Removes use of any colors aside from black and white so we don’t get errors when we inevitably replace core Tailwind colors in a project

coreymcollins committed:
Adds a function to extras for grabbing attachment IDs

coreymcollins committed:
Adds Tailwind to the readme

coreymcollins committed:
Cleans up and reorders the config a little bit

coreymcollins committed:
Cleans up some linting issues

coreymcollins committed:
Removes console statements

coreymcollins committed:
Replaces quotation marks

coreymcollins committed:
Ignores our index.js file from linting

coreymcollins committed:
Fixes some linting errors

coreymcollins committed:
Cleans up some linting in the Tailwind config

coreymcollins committed:
Ignores a couple of JS files

coreymcollins committed:
Fixes a dangler

coreymcollins committed:
Adds some items to make theme check happy

coreymcollins committed:
Fixes a couple of theme check complaints

aubreypwd committed:
Integrate Coding Standards Linting, Auto-formatting, & Husky. (#553)

gregrickaby committed:
dependency bump

gregrickaby committed:
opt-in to future changes

aubreypwd committed:
Ignore dist/ when linting.

aubreypwd committed:
Ignore dist/

aubreypwd committed:
npm update chores

gregrickaby committed:
bump tailwind to 1.9.3

aubreypwd committed:
Tell VSCode not to validate SCSS

Bandaid so that warning from VSCode do not apply when we use Tailwind syntax in SCSS.

aubreypwd committed:
Merge branch 'feature/tailwind' of ssh://github.com/WebDevStudios/wd_s into feature/tailwind

aubreypwd committed:
Ignore dist/

gregrickaby commented:
**Here's my release plan for 2.0.**

The `feature/tailwind` branch will be renamed to `main`. I'll update the default branch of this repository to `main`. I'll also tag and release a `2.0` off the `main` branch.

**Why not just merge into `master`?**

There are too many changes to the 2.0 version to just merge into `master`. We also don't need 5+ years of commit history, since 2.0 is so different. The current `master` branch will be renamed to `pre-tailwind` and kept around for historical purposes.

coreymcollins committed:
Safe guards the html font size for any plugins that override

coreymcollins committed:
Merge branch 'feature/tailwind' of github.com:WebDevStudios/wd_s into feature/tailwind
